crates/bevy_dylib/Cargo.toml

```diff
@@ -12,7 +12,9 @@ keywords = ["bevy"]
 crate-type = ["dylib"]
 
 [dependencies]
-bevy_internal = { path = "../bevy_internal", version = "0.16.0-dev", default-features = false }
+bevy_internal = { path = "../bevy_internal", version = "0.16.0-dev", default-features = false, features = [
+    "std",
```

Comment:
It is worth documenting why we're requiring `std` here. I doubt future people will be able to intuit that this fixes an obscure linking bug in a specific crate.